### PR TITLE
MPCFill tags

### PR DESCRIPTION
**Pull request by glandu-sr**
First draft of tag implementation for MPCFill.

Tags are found from filename, between parentheses or brackets.
The list of knowns tags is an enum in the Tag class, with multiple values possible.

If this implementation is validated, I'll look into how to render the tags on the website.

### CODE REVIEW

**Comment by glandu-sr** [comment]
I did not make any migration yet, so it is expected that the tests do not pass.

**Comment by glandu-sr** [comment]
I'm not sure. It could be in either autofill (website) or MPCAutofill (database/backend) as it will be used in both.

**Review by ndepaola — COMMENTED** [review]
agreed that this looks promising! just a couple of thoughts after a brief skim

File: MPCAutofill/cardpicker/tags.py
Comment (glandu-sr) [reply MPCAutofill/cardpicker/tags.py:20]: The problem is that `(`, `)`, `[` and `]` need to be escaped which hurts readability.
`\(([^\(\)]+)\)` is to match the contents of `(...)` and `\[([^\[\]]+)\]` the contents of `[...]`.
We could just accept one of the two for the tags but I've seen both in the file names on the drives.
But if you have a better idea for it, I'm open to it.

File: MPCAutofill/cardpicker/models.py
Comment (glandu-sr) [reply MPCAutofill/cardpicker/models.py:100]: Didn't know arrayfield, will update it when I have time.

**Comment by glandu-sr** [comment]
I've updated the merge request to use ArrayField and set the default engine to postgres.
But it seems to break the tests as the action does not setup a postgres server.
I also would like to add tests to see if the tags work as expected but not really sure how to go about it.

@ndepaola I suggest to merge this one first, to see if it does not break anything, before integrating it to the frontend. If you find nothing to change of course.
Tell me if you want me to search how to fix the action or if you want to do it yourself.

**Comment by ndepaola** [comment]
sweet thanks, this looks good to me! i'll test it a bit locally when i have time before merging.

side note - do you happen to know if it's possible to migrate a database's contents from sqlite to postgres? doesn't matter to me personally but it might for other people using this project

**Comment by glandu-sr** [comment]
https://sweetcode.io/django-postgresql-migration-from-sqlite/
This article seem to have the commands to use for the migration.

**Comment by ndepaola** [comment]
very sorry for forgetting about this and letting it get stale. could you please rebase this on master when you have time? i'll merge it as-is and flesh out tests/migrations/etc afterwards

**Comment by glandu-sr** [comment]
No problem, you were doing the rework of the frontend which seems to have been a lot of work.  
The rebase is done, I had some conflicts, mainly for dependencies so I hope it broke nothing ^^

**Comment by glandu-sr** [comment]
The github action is still broken by this merge request.  
For the Tag enum, it may become massive, so not really sure if it is the best here. But no idea what other alternative we may have. A Tags table in the database ? 

List of tags that may be added off the top of my head :
- frame type (classic, old border, minimalist, showcase, ...). Maybe a custom frame category too ?
- AI art
- language (JP, FR, DE, IT, ...)
- alternative name
- other IP ? (Yu-Gi-Oh, Pokémon, Flesh & Blood, ...). It may go in the alt name category.

With a custom card tag, it may be possible to later have a custom card browsing page for cards created by the community.

**Comment by ndepaola** [comment]
> No problem, you were doing the rework of the frontend which seems to have been a lot of work. The rebase is done, I had some conflicts, mainly for dependencies so I hope it broke nothing ^^

excellent, thank you for that!!

> The github action is still broken by this merge request. For the Tag enum, it may become massive, so not really sure if it is the best here. But no idea what other alternative we may have. A Tags table in the database ?
> 
> List of tags that may be added off the top of my head :
> 
>     * frame type (classic, old border, minimalist, showcase, ...). Maybe a custom frame category too ?
> 
>     * AI art
> 
>     * language (JP, FR, DE, IT, ...)
> 
>     * alternative name
> 
>     * other IP ? (Yu-Gi-Oh, Pokémon, Flesh & Blood, ...). It may go in the alt name category.
> 
> 
> With a custom card tag, it may be possible to later have a custom card browsing page for cards created by the community.

the github actions being broken is fine - this happens because there are tests for the google drive integration, and google drive auth is set up with repository secrets - CI running on forks don't pick up the auth configured in the parent repo so the tests will be failing because of that. no stress, if tests are broken after this is merged i'll fix them

you read my mind w/ having a tag table in the database - will give it some more thought :)

**Review by ndepaola — APPROVED** [review]

**Pull request merged**
